raft/src/main/resources/common/message/QuorumStateData.json

```diff
@@ -16,10 +16,10 @@
 {
   "type": "data",
   "name": "QuorumStateData",
-  // Version 1 removes clusterId field.
-  "validVersions": "0-1",
+  "validVersions": "0",
   "flexibleVersions": "0+",
   "fields": [
+    {"name": "ClusterId", "type": "string", "versions": "0+"},
```

Comment:
I am confused, why not just have ClusterId and `versions: "0"` (no plus sign at the end)?

You can't change the past, but you certainly can create a new version without this field if you really want.

Reply:
This PR reverts a commit that made it to `trunk` but it is not included in any of the releases. We need to revert this before we can release 3.5.

> I am confused, why not just have ClusterId and versions: "0" (no plus sign at the end)?

I think that that may be part of the change. We still want to handle then software downgrade since 3.4 and less assumes `clusterId` exists. In the associated Jira I suggested creating a KIP for this improvement.

Reply:
Changing the version to "0" cannot fix this backward compatibility issue. The version is designed during writing the state file with highest supported version. So, once we write the file with a newer version format, the downgrade will fail. Agree to revert the change first, and then decide if we want to raise a KIP to fix it.

Reply:
Yeah. Maybe we can do this in 4.0 but I need to confirm that 4.x won't support software downgrade to 3.x.